### PR TITLE
Add bulk upload

### PR DESCRIPTION
**Pull request by oliverroick**
Adds initial UI components to enable uploading ZIP archives for bulk processing. 

Major changes:

- Added a new button and modal for the upload, including a form
- Implemented the upload (against a fake postman API for now, while Nick is working on the backend implementation)
- Added upload progress tracking and a progress bar. It used XMLHttpRequest because fetch doesn't allow for tracking progress. I'm also aware that we might have to make some changes to the implementation once the upload API is in place. 

Opening this as a draft PR for to get some initial feedback, @nathanielrindlaub. Specifically looking for notes on reducers and thunks. I placed them in the project slice, but I'm not sure this is the right place. Should this go into a separate slice?

### CODE REVIEW

**Review by nathanielrindlaub — COMMENTED** [review]
This all looks great @oliverroick & the additions you've made to the `projects` state slice are perfect. The only thing I'd suggest before getting too far down this path is to break the upload stuff into it's own slice of state (rather than part of `projects` - that one is getting a little unwieldy as it is). 

To do that, create a new `src/features/uploads` directory and add a `uploadsSlice.js` (you could just copy past `projectsSlice.js` right in and strip out the non-upload stuff), and then add the exported reducer to `src/app/rootReducer.js`. Does that make sense?

Also, this is up to you but I think in the `projectSlice.js` I originally nested all the various entities that needed fetching (`projects`, `views`, etc) in a `loadingStates` object because there were a bunch of different async things being loaded and tracked in that one slice... if you pull the `upload` state into its own slice you could probably ditch the `loadingStates` level and flatten it out a bit. I'll leave the state shape up to you though do whatever makes sense to you.

**Comment by oliverroick** [comment]
@nathanielrindlaub this is now working with the upload to S3. Could you review when you have a moment?

**Review by nathanielrindlaub — COMMENTED** [review]
This all looks great @oliverroick! I haven't tested it yet but the code looks good to me!

File: src/features/upload/uploadSlice.js
Comment (nathanielrindlaub) [inline src/features/upload/uploadSlice.js:1]: @oliverroick - this is all looking great. Only question I have is why in the `fetchBatches` thunk it looked like you had at first filtered the returned batches from `getBatches` and only fetched the detail (`fetchBatch`) for the ongoing ones, but that filtering is now commented out. Any reason to keep it commented or should we add that filtering back in?

**Review by nathanielrindlaub — COMMENTED** [review]
Just two small comments but otherwise looking great @oliverroick! Just about ready to merge I recon.

File: src/features/upload/BulkUploadForm.js [hunk 1]
```diff
@@ -1,25 +1,128 @@
...
+          <tr>
+            <TableHeadCell>File name</TableHeadCell>
+            <TableHeadCell>Status</TableHeadCell>
+            <TableHeadCell>Actions</TableHeadCell>
+          </tr>
+        </thead>
+        <tbody>
+          {sortedBatchStates.map((batch) => {
+            const { _id, originalFile, processingEnd, total, remaining } = batch
+            const isStopable = !processingEnd && (remaining === null || total - remaining > 0);
```
[the remark this answers is not in the document]
Comment (oliverroick) [reply src/features/upload/BulkUploadForm.js:108]: I was assuming you should only be able to stop the batch if there are still images left to process. If the remaining images is 0 and `processingEnd` is `null` then the processing is being finalised so I figured at that stage you shouldn't be able to stop the process anymore.

I can change the condition to `>=0` — @ingalls what happens when you stop the process after all images have been processed. I'm guessing there's still some work happening in the background like tearing down queues or whatever. Is it save the stop the batch at this point?

File: src/features/upload/uploadSlice.js
Comment (oliverroick) [reply src/features/upload/uploadSlice.js:1]: Initially we would only get individual batch information from ongoing batches to update the `remaining` value. But to get the errors as well, we need to get the details for _all_ batches as they errors are currently only exposed in the `GetBatch` request

Comment (nathanielrindlaub) [reply src/features/upload/uploadSlice.js:1]: ah, gotcha. Ok let's leave it as is!

File: src/features/upload/BulkUploadForm.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ingalls) [reply src/features/upload/BulkUploadForm.js:113]: @oliverroick It's safe to call StopBatch at any point in the process. It just fires up the IngestDelete lambda which is the exact same lambda that the cloudwatch alarms would call. As such if the stack is already terminated, or in the process of terminating it wouldn't harm anything.

**Pull request merged**